astroquery/nist/tests/test_nist.py

```diff
@@ -45,6 +45,13 @@ def test_query_async(patch_get):
                                           linename="H I", get_query_payload=True)
     assert response['spectra'] == "H I"
     assert response['unit'] == nist.core.Nist.unit_code['nm']
+    response = nist.core.Nist.query_async(4000 * u.nm, 7000 * u.nm,
+                                          linename=["H I", "Fe I"], get_query_payload=True)
+    assert response["spectra"] == "H I Fe I"
+    with pytest.raises(TypeError) as err:
+        nist.core.Nist.query_async(4000 * u.nm, 7000 * u.nm,
+                                          linename=("H I", "Fe"), get_query_payload=True)
```

Comment:
Only reason it's here is because I only added the support for lists. I could add support for tuples if or other dtypes if its desired. I only ended up doing lists cause when looking at the vectorizing with vizier it looked like only lists were supported.

Reply:
I would think any iteratives are fair game here, though generators maybe a bit going too far :)